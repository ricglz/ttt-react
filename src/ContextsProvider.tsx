import React, { useState, useCallback } from 'react';
import { IntlProvider } from 'react-intl';
import { NotificationContainer } from 'react-notifications';
import { BrowserRouter as Router } from 'react-router-dom';
import messages from './messages';

type Locale = string;

function getDirection(locale: Locale) {
  return locale === 'ar' ? ['rtl', 'text-right'] : ['ltr', 'text-left'];
}

type ChildrenComponentProps = {
  locale: Locale;
  changeLocale: (locale: Locale) => void;
};

type Props = {
  ChildrenComponent: React.FC<ChildrenComponentProps>;
  RouterComponent?: any;
};

export default function ContextsProvider({
  ChildrenComponent,
  RouterComponent = Router,
}: Props) {
  const [locale, setLocale] = useState<Locale>('en');

  const changeLocale = useCallback((newLocale) => {
    setLocale(newLocale);
  }, []);

  const [direction, klass] = getDirection(locale);

  return (
    <IntlProvider locale={locale} messages={messages[locale]}>
      <div dir={direction} className={klass}>
        <RouterComponent>
          <ChildrenComponent changeLocale={changeLocale} locale={locale} />
        </RouterComponent>
        <NotificationContainer />
      </div>
    </IntlProvider>
  );
}
