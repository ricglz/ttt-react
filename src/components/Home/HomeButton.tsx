import React from 'react';
import { FormattedMessage } from 'react-intl';
import { Link } from 'react-router-dom';

function renderText(staticText: boolean, text: string) {
  return staticText ? <span>{text}</span> : <FormattedMessage id={text} />;
}

type Props = {
  text: string,
  staticText?: boolean,
  url: string,
};

const HomeButton = ({ text, staticText = false, url }: Props) => (
  <div className="row justify-content-center border-top py-3">
    <Link className="btn btn-home" to={url}>{renderText(staticText, text)}</Link>
  </div>
);

export default HomeButton;
