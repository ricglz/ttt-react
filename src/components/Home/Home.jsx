import React from 'react';
import { FormattedMessage } from 'react-intl';
import HomeButton from './HomeButton';

function Home() {
  return (
    <div className="container text-center">
      <div className="row">
        <div className="col">
          <FormattedMessage id="homePage.title" default="Home Page">
            {txt => <h1>{txt}</h1>}
          </FormattedMessage>
        </div>
      </div>
      <HomeButton text="shared.sp" url="/singleplayer" />
      <HomeButton text="shared.mp" url="/multiplayer" />
      <HomeButton staticText text="Online" url="/online" />
      <HomeButton text="shared.tutorial" url="/tutorial" />
      <HomeButton staticText text="Contributors" url="/contributors" />
    </div>
  );
}

export default Home;
