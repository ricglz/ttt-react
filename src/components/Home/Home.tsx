import React from 'react';
import { FormattedHeader } from '../Layout/FormattedText';
import HomeButton from './HomeButton';

function Home() {
  return (
    <div className="container text-center">
      <div className="row">
        <div className="col">
          <FormattedHeader locale="homePage.title" defaultMessage="Home Page" />
        </div>
      </div>
      <HomeButton text="shared.sp" url="/singleplayer" />
      <HomeButton text="shared.mp" url="/multiplayer" />
      <HomeButton staticText text="Online" url="/online" />
      <HomeButton text="shared.tutorial" url="/tutorial" />
      <HomeButton staticText text="Contributors" url="/contributors" />
    </div>
  );
}

export default Home;
