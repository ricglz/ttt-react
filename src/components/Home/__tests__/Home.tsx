import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Home from '../Home';
import ContextsProvider from '../../../ContextsProvider';

function Component() {
  return (
    <ContextsProvider ChildrenComponent={Home} RouterComponent={MemoryRouter} />
  );
}

it('renders without crashing', () => {
  const component = <Component />;
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});
