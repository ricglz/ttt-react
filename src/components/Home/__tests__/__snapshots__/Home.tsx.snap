// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="text-left"
  dir="ltr"
>
  <div
    className="container text-center"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <h1>
          Home Page
        </h1>
      </div>
    </div>
    <div
      className="row justify-content-center border-top py-3"
    >
      <a
        className="btn btn-home"
        href="/singleplayer"
        onClick={[Function]}
      >
        Single Player
      </a>
    </div>
    <div
      className="row justify-content-center border-top py-3"
    >
      <a
        className="btn btn-home"
        href="/multiplayer"
        onClick={[Function]}
      >
        Local Multiplayer
      </a>
    </div>
    <div
      className="row justify-content-center border-top py-3"
    >
      <a
        className="btn btn-home"
        href="/online"
        onClick={[Function]}
      >
        <span>
          Online
        </span>
      </a>
    </div>
    <div
      className="row justify-content-center border-top py-3"
    >
      <a
        className="btn btn-home"
        href="/tutorial"
        onClick={[Function]}
      >
        How to play
      </a>
    </div>
    <div
      className="row justify-content-center border-top py-3"
    >
      <a
        className="btn btn-home"
        href="/contributors"
        onClick={[Function]}
      >
        <span>
          Contributors
        </span>
      </a>
    </div>
  </div>
  <div
    className="notification-container notification-container-empty"
  >
    <div />
  </div>
</div>
`;
